src/Core/Services/CryptoService.cs

```diff
@@ -997,6 +1002,32 @@ private async Task<TKey> MakeKeyAsync<TKey>(string password, string salt, KdfCon
             return keyCreator(key);
         }
 
+        private async Task<bool> ValidateUserKeyAsync(UserKey key, string userId = null)
+        {
+            if (key == null)
+            {
+                return false;
+            }
+
+            try
+            {
+                var encPrivateKey = await _stateService.GetPrivateKeyEncryptedAsync(userId);
+                if (encPrivateKey == null)
+                {
+                    return false;
+                }
+
+                var privateKey = await DecryptToBytesAsync(new EncString(encPrivateKey), key);
+                await _cryptoFunctionService.RsaExtractPublicKeyAsync(privateKey);
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
⛏️ No need for `(Exception)` if you just want to catch everything and IMO it's better to return inside the try so when reading the code you don't need to skip or jump to know what's next just for returning:
```suggestion
                await _cryptoFunctionService.RsaExtractPublicKeyAsync(privateKey);
                return true;
            }
            catch
            {
                return false;
            }
```